src/jvm/main/org/jetbrains/kotlinx/lincheck/Lincheck.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck
+
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionResult
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionScenario
+import org.jetbrains.kotlinx.lincheck.execution.parallelResults
+import org.jetbrains.kotlinx.lincheck.strategy.LincheckFailure
+import org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedCTestConfiguration
+import org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedStrategy
+import org.jetbrains.kotlinx.lincheck.strategy.managed.forClasses
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck.strategy.verify
+import org.jetbrains.kotlinx.lincheck.transformation.LincheckJavaAgent.ensureObjectIsTransformed
+import org.jetbrains.kotlinx.lincheck.transformation.withLincheckJavaAgent
+import org.jetbrains.kotlinx.lincheck.verifier.Verifier
+
+
+/**
+ * This method will explore different interleavings of the [block] body and all the threads created within it,
+ * searching for the first raised exception.
+ *
+ * @param invocations number of different interleavings of code in the [block] that should be explored.
+ * @param block lambda which body will be a target for the interleavings exploration.
+ */
+fun <R> runConcurrentTest(
+    invocations: Int = DEFAULT_INVOCATIONS_COUNT,
+    block: () -> R
```

Comment:
Please add a test on Java, with and without `invocations`,  to check the API.

Reply:
Unfortunately, using this API from Java is currently unsupported due to technical limitations:
https://github.com/JetBrains/lincheck/issues/445

Reply:
Yes, so let's ignore the test. However, we need to see what the code looks like in Java.

Reply:
Note: as we are not working on the API now, we can postpone this question.